### ISSUE TITLE
Task: configure student pre-survey by 1/9 (20,5,4)

### ISSUE THREAD

**Issue by stenington** [Iridescent-CM/technovation-app#2330]
Student pre-surveys need to go up no later than 1/9.

This ticket is to work out the configuration plan for that, including Surveymonkey custom variable integration, completion tracking on the platform if necessary, admin field values, and even a monitoring plan if we want to be that thorough.

**Comment by rsgonzal** [comment]
Option 1:
- Reset the platform, so even returning users are asked to complete

Option 2:
- Reset the platform, so even returning users are asked to complete
- Set up the survey collector url with query parameters to collect all the data points outlined in #2310 
- Rebecca will make a version of the survey without the fields and we can test to see if the information is being passed

Option 3:
- Reset the platform, so even returning users are asked to complete
- Set up the survey collector url with query parameters to collect all the data points outlined in #2310 
- Rebecca will make a version of the survey without the fields and we can test to see if the information is being passed
- Provide the user ID from the platform

**Comment by stenington** [comment]
Implicit in all options, in my mind, is that someone also configures the link, main text, and optional text in the admin. For option 1, the modal will be shown to everyone twice, even if they took the survey the first time it was shown, so the text could include a thank you for those who have already taken the survey.

**Comment by hellafitz** [comment]
Decision: do Option 2 + provide User ID from the platform. @stenington to make a separate ticket for changing survey frequency

**Comment by stenington** [comment]
Ok, so the plan per decisions above is:

* Reset the platform, so even returning students are asked to complete 
* Set up the survey collector url with query parameters to collect all the data points outlined in #2310, plus `uid_value` after #2338 is live
* Rebecca will make a version of the survey without the fields and we can test to see if the information is being passed
* Completion tracking set up, if they've completed the survey the pop up shouldn't show up

The [documentation here](https://docs.google.com/document/d/1hVaugqKipnD8pYmCcMl5KWCHrrqBBSr-86MQZTdm5P8/edit#) needs to be updated with instructions for resetting the modal, and instructions on completion tracking.

**Comment by stenington** [comment]
@rsgonzal How much of this do you think needs testing in QA? I'd like a quick check of the new `uid_value` parameter. What else do you want to check in staging?

**Comment by rsgonzal** [comment]
I think we should do the testing as if it were production. Does this make sense?

- Get a version of the survey with less fields
- Set up the query parameters (this is done in survey monkey?)
- Pull data from survey monkey to make sure data is being sent after the survey is taken

Student - new one and old one
- take the survey
- sign out and sign in a few times to make sure you're not asked to take again

**Comment by stenington** [comment]
Both parts make sense, but with caveats.

For the first part, we should also configure the custom end page to redirect back to the platform and mark the survey as complete.

For the second part, with the old student we'll need to coordinate resetting the pop ups on staging first (which is a good thing to do). Then signing out and signing back in a few times won't really test whether you see the pop up again, as it has a 2 day "cool down" period. We'd either have to wait two or three days and then log back in, or we can maybe look in the database to confirm that the survey completion was recorded and then trust that that will prevent the pop up from showing again.

With the new student, the account has to be two days old before the pop up shows at all, so again we'd either need to create the account, then come back in two or three days, or we'd need to maybe tweak their sign up date in the database or something? 

Documentation for [exactly how the pop up logic works is here](https://docs.google.com/document/d/1hVaugqKipnD8pYmCcMl5KWCHrrqBBSr-86MQZTdm5P8/edit#heading=h.2v0cxcs7ze3y) for reference.

**Comment by rsgonzal** [comment]
Ok, here is the new survey. Do the variables get added in survey monkey?

https://www.surveymonkey.com/summary/R7NE7LYlohHmajs4DYZ7VvGqHL_2FdW0hYabbMld_2Fbq4ZgvUdeLDPxI9CWBmUMo1Ul?ut_source=my_surveys_list

**Comment by rsgonzal** [comment]
Actually, reading your documentation doc now!

**Comment by rsgonzal** [comment]
Here is what I've done so far:
1. Got a version of the survey without extra fields: '2020 TG Student Pre-Survey WITHOUT Repeat Questions'
2. Checked it had these custom variables:
- user id as [uid_value] **had to add this one**
- email as [email_value]
- country as [country_value]
- state or province as [state_value]
- user's full name as [name_value]
- city as [city_value]
- age as [age_value]
- school or company name as [school_value] or as [company_value]
- list of team names as [team_value]
3. Created a new collector link: https://www.surveymonkey.com/r/PD6XXS3?name=[name_value]&school=[school_value]&city=[city_value]&state=[state_value]&country=[country_value]&email=[email_value]&team=[team_value]&age=[age_value]&uid=[uid_value]
4. Verified it had the tokens^^
5. Logged in as roxana+11@iridescentlearning.org
- did not immediately see a popup
6. Went to more information and followed survey link there and took it
7. Downladed data through an export on survey monkkey: https://docs.google.com/spreadsheets/d/1gxXVzf8OcgPpsg90NMZG9Br83HDE4a_jgDGLwWKCNwY/edit#gid=1775522975

It looks like the data was passed through. I'm not sure what the first three name/email fields are but the ones from the platform are on the last colums. How can I verify 351 is the correct user id for roxana+11@iridescentlearning.org @stenington ?

**Comment by rsgonzal** [comment]
I've now done these:
1. Set the custom end page to: https://my.technovationchallenge.org/survey_completion

![]()
2. Logged in as roxana+90@iridescentlearning.org
3. I did see the pop up to take the survey and clicked on it
4. unfortunately it told me i had already taken the survey

Should i set the survey to allow multiple responses?

**Comment by rsgonzal** [comment]
1. Logged in as rsgonzal10+11@gmail.com
2. Saw the popup and clicked on the link
3. took the survey
4. redirect didn;t work because it took me to production

**Comment by rsgonzal** [comment]
1. Changed custom end page to: https://technovation-qa.herokuapp.com/survey_completion
2. Logged in as roxana+00@iridescentlearning.org
3. got the popup and clicked the link
4. completed the survey
5. got the thank you page, clicked done
6. Was redirected to platform and banner saying thanks for taking survey popped up

is this the only one we can check to see if it was recorded @stenington

**Comment by rsgonzal** [comment]
Would it be better if there wasn't the thank you page? I feel like people might not click 'done' and could just exit out of it

**Comment by stenington** [comment]
> How can I verify 351 is the correct user id for roxana+11@iridescentlearning.org

You can look up that user in the admin and check the url, which has the id. It's 351, so looks like the user id got passed correctly.

> Should i set the survey to allow multiple responses?

If you don't, people will only be allowed to take the survey once **per computer**, basically. So if there might be situations where multiple users share a computer (like a computer lab or something) then yes, you should turn on multiple responses.

> Other than that, I'm pretty confident about the completion tracking since it goes away in the more info section, what do you think?

Your call. I think we can probably expect that it won't work for some people, but I don't know how many. Since Devin was okay with some people continuing to see the pop up in the other ticket, I don't see why it would be different here.

**Comment by stenington** [comment]
Ok, so let's work on a deployment plan. I'm seeing something like:

1. confirm survey configuration
    * custom variables
    * multiple responses ON
    * no thank you page?
    * end page redirect pointing to production
2. make sure **no survey** is configured in production
3. developer resets the variables that track pop ups and survey completion for the relevant users
    * for this ticket, that's all student accounts
4. admin configures survey
    * link includes the query parameters
    * text includes a "thanks if you've taken this already" message of some sort

**Comment by stenington** [comment]
@rsgonzal read over the suggested plan in the previous comment and let me know if you see something i've forgotten please!

**Comment by rsgonzal** [comment]
Ok, I took away the last thank you page - Rebecca confirmed this was ok. I also allowed for multiple responses.

I think this is a good plan! Bringing up in the Technovation meeting that I need help with the pop up text and testing in production.

Do you know why these first, last, and email columns pop up at the beginning? Rebecca and I noticed them. The ones transferred from the site are at the end.

![]()

**Comment by stenington** [comment]
I don't know off the top of my head why those columns are in there.

**Comment by rsgonzal** [comment]
ok! not a big deal!

So for our plan:
1. confirm survey configuration: https://www.surveymonkey.com/r/PD6XXS3?name=[name_value]&school=[school_value]&city=[city_value]&state=[state_value]&country=[country_value]&email=[email_value]&team=[team_value]&age=[age_value]&uid=[uid_value]
    * custom variables
    * multiple responses ON
    * no thank you page?
    * end page redirect pointing to production
2. make sure **no survey** is configured in production

**Comment by rsgonzal** [comment]
@stenington this is ready for 3.

3. developer resets the variables that track pop ups and survey completion for the relevant users
for this ticket, that's all student accounts

**Comment by stenington** [comment]
@rsgonzal Cool, I reset that data and the uid code has been deployed, so we should be clear for step 4 whenever you want the surveys to go back up.

**Comment by rsgonzal** [comment]
Awesome! Step 4 is done, tested in production and pulled the data from surveymonkey. All is working as expected.
